packages/core/src/Ignition.ts

```diff
@@ -227,7 +234,89 @@ export class Ignition {
 
     const { executionGraph } = transformResult;
 
-    return { deploymentGraph, executionGraph };
+    return {
+      deploymentGraph,
+      executionGraph,
+      networkName: networkNameByChainId[chainId] ?? "unknown",
+    };
+  }
+
+  public async info(moduleName: string): Promise<ModuleInfoData[]> {
+    log(`Start info`);
+
+    const journalData: {
+      [networkTag: string]: Array<
+        DeployStateExecutionCommand & { chainId: number }
+      >;
+    } = {};
+    for await (const command of this._journal.readAll()) {
+      const network = networkNameByChainId[command.chainId] ?? "unknown";
+      const networkTag = `${network}:${command.chainId}`;
+
+      if (journalData[networkTag] === undefined) {
+        journalData[networkTag] = [];
+      }
+
+      journalData[networkTag].push(command);
+    }
+
+    const deployments: Deployment[] = [];
+    for (const [networkTag, commands] of Object.entries(journalData)) {
+      const deployment = new Deployment(
+        moduleName,
+        this._services,
+        this._journal
+      );
+
+      const [networkName, chainId] = networkTag.split(":");
+      await deployment.setDeploymentDetails({
+        networkName,
+        chainId: +chainId,
```

Comment:
In the same situation, I would do a `parseInt`. This is the sort of thing that should be an eslint rule, but I am will ping the typescript channel and see what the collective wisdom is.

Reply:
Alright, the channel has spoken (Franco) and it is neither, `Number(chainId)` wins!

Reply:
The channel speaks again. `parseInt` for anything that might come from a user, internally we don't care. Given this is entirely internal, I think it can be left as it is.